### PR TITLE
Add E2E test for lock contention flags

### PR DESCRIPTION
**Pull request by ipochi**
Adds a new job ci-kubernetes-node-kubelet-lock-contention to the test suite.

This test is created due to the need of `--lock-file` and `--exit-on-lock-contention` 
be moved as flags and into the Kubelet configuration file rather than be dropped.

Adds a new tab in TestGrid named `kubelet-gce-e2e-lock-contention`
running tests focused on `NodeSpecialFeature:LockContention`.

This test was earlier added in the PR: https://github.com/kubernetes/test-infra/pull/22758

However it was later removed in this [commit](https://github.com/kubernetes/test-infra/commit/1b54c26a6ec7e483dcc4c07e81ae96c71a08b269#diff-14f46258d64962b171c1662b99c4e01ea48674ca3ef00b1d9dec13acf2ad91f9)

Corresponding e2e test is at https://github.com/kubernetes/kubernetes/pull/104334

Signed-off-by: Imran Pochi <imran@kinvolk.io>

### CODE REVIEW

**Comment by elfinhe** [comment]
seems like all node serial tests will be running with this flag. Is that what we want? btw, pull-kubernetes-node-kubelet-serial and pull-kubernetes-node-kubelet-serial-containerd are not triggered. So we don't know if it will affect existing serial tests. 

maybe we can wait until Monday then do a quick verify, or it may be possible to update run_if_changed to trigger the serial tests to see the results.

**Comment by ipochi** [comment]
> seems like all node serial tests will be running with this flag. Is that what we want? btw, pull-kubernetes-node-kubelet-serial and pull-kubernetes-node-kubelet-serial-containerd are not triggered. So we don't know if it will affect existing serial tests.
> 
> maybe we can wait until Monday then do a quick verify, or it may be possible to update run_if_changed to trigger the serial tests to see the results.

I don't there should be any impact on the existing tests, but it would be a good to trigger the serial tests based on the new change.

I had a question in the corresponding PR, is the question valid and possible ? https://github.com/kubernetes/kubernetes/pull/104334#issuecomment-898280645

**Comment by ehashman** [comment]
/hold

I don't think we want to run all serial tests with this feature enabled... We discussed in SIG Node CI subproject a while back, not sure if there's been any progress there.

**Comment by ipochi** [comment]
> /hold
> 
> I don't think we want to run all serial tests with this feature enabled... We discussed in SIG Node CI subproject a while back, not sure if there's been any progress there.

@ehashman Update the test job to reflect the correct string `NodeSpecialFeature` and added skip section to avoid the test being picked up by `Flaky` or `Serial` test suite.

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml
Comment (ehashman) [inline config/jobs/kubernetes/sig-node/node-kubelet.yaml:459]:
```suggestion
          - --test_args=--nodes=1 --focus="\[NodeSpecialFeature:LockContention\]" --skip="\[Flaky\]"
```

If these tests need to be run serially, we should include that on them.

**Review by ehashman — COMMENTED** [review]
/hold cancel

We should first merge https://github.com/kubernetes/kubernetes/pull/104334 before merging this in case of future changes.

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml
Comment (ipochi) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml]: This test need not be run serially, as the test is disruptive and restarts Kubelet.

**Comment by ipochi** [comment]
@endocrimes Updated this PR (title and description as well) to bring back the E2E test as per (your comment)[https://github.com/kubernetes/kubernetes/pull/104334#issuecomment-1040551089].

/cc @ehashman @SergeyKanzhelev

**Review by ehashman — COMMENTED** [review]
/approve
/lgtm

/hold 
for the test PR to merge first

**Review by SergeyKanzhelev — APPROVED** [review]
/unhold

**Pull request merged**
